[AIRFLOW-142] setup_env.sh doesn't download hive tarball if hdp is specified as distro

sekikn opened the pull request:
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- AIRFLOW-142

While both Hadoop and Hive tarballs are downloaded if cdh is specified
as "distro" parameter, only Hadoop is downloaded if hdp is specified.
This patch fixes this problem and updates HDP version to more recent one,
which is consistent with the version that minicluster-1.1 uses.

bolkedebruin commented on scripts/ci/setup_env.sh:
```diff
@@ -70,6 +70,7 @@ if [ $HADOOP_DISTRO = "cdh" ]; then
     HIVE_URL="http://archive.cloudera.com/cdh5/cdh/5/hive-latest.tar.gz"
 elif [ $HADOOP_DISTRO = "hdp" ]; then
     URL="http://public-repo-1.hortonworks.com/HDP/centos6/2.x/updates/2.0.6.0/tars/hadoop-2.2.0.2.0.6.0-76.tar.gz"
+    HIVE_URL="http://public-repo-1.hortonworks.com/HDP/centos6/2.x/updates/2.0.6.0/tars/hive-0.12.0.2.0.6.0-76.tar.gz"
```
Can you please use a more recent hive? Hive 1.2.1 is stable in HDP Hive 2.0.0 soon. 0.12 is really old.

sekikn committed:
[AIRFLOW-142] setup_env.sh doesn't download hive tarball if hdp is specified as distro

While both Hadoop and Hive tarballs are downloaded if cdh is specified
as "distro" parameter, only Hadoop is downloaded if hdp is specified.
This patch fixes this problem and updates HDP version to more recent one,
which is consistent with the version that minicluster-1.1 uses.
